### PR TITLE
*: update document about auto-increment column must be key constraint

### PR DESCRIPTION
**Pull request by tiancaiamao**
### First-time contributors' checklist 

- [X] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

As the title says.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [X] master (the latest development version)
- [X] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/42305

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: auto-increment.md [hunk 1]
```diff
@@ -351,7 +351,7 @@ CREATE TABLE t(a int AUTO_INCREMENT key) AUTO_ID_CACHE 1;
 
 目前在 TiDB 中使用 `AUTO_INCREMENT` 有以下限制：
 
-- 定义的列必须为主键或者索引的首列。
+- 定义的列必须为主键或者索引的首列（TiDB 7.0.0 之后去掉了该限制）。
```
Comment (ran-huang) [inline auto-increment.md:354]: 在 7.0 的文档里，这一条可以删掉吧。或者改成这样：
```suggestion
- 对于 v6.6.0 及更早的 TiDB 版本，定义的列必须为主键或者索引的首列。
```

File: mysql-compatibility.md [hunk 2]
```diff
@@ -67,21 +69,28 @@ TiDB 高度兼容 MySQL 5.7 协议、MySQL 5.7 常用的功能及语法。MySQL
...
-|           5 |    2 |
-|           6 |    3 |
+|           2 |    1 |
+|           4 |    3 |
+|           6 |    5 |
 +-------------+------+
 3 rows in set (0.01 sec)
 ```
 
+> 可以看到，由于共用分配器，id 每次自增是 2。[MySQL 兼容模式](/auto-increment.md#mysql-兼容模式) 中改掉了该行为，没有共用分配器，不会跳号。
```
Comment (ran-huang) [inline mysql-compatibility.md:92]:
```suggestion
可以看到，由于共用分配器，id 每次自增步长是 2。在 [MySQL 兼容模式](/auto-increment.md#mysql-兼容模式)中改掉了该行为，没有共用分配器，因此不会跳号。
```

File: auto-increment.md (hunk 1 above)
Comment (benmaoer) [reply auto-increment.md:354]: 支持，@ran-huang  的表述更清晰

File: mysql-compatibility.md [hunk 3]
```diff
@@ -59,6 +59,8 @@ TiDB 高度兼容 MySQL 5.7 协议、MySQL 5.7 常用的功能及语法。MySQL
 
 自增 ID 详情可参阅 [AUTO_INCREMENT](/auto-increment.md)。
 
+- TiDB v7.0.0 之后不再要求自增列必须带索引。MySQL 5.7 以及更早版本使用 InnoDB 时会有该约束，而 MySQL 8.0 中也去掉了该约束。[#40580](https://github.com/pingcap/tidb/issues/40580)
```
Comment (benmaoer) [inline mysql-compatibility.md:62]:
```suggestion
- 对于 v6.6.0 及更早的 TiDB 版本，TiDB 的行为与 MySQL InnoDB  保持一致，要求自增列必须为主键或者索引前缀。 从 v7.0.0 开始，TiDB 移除自增列必须是索引或索引前缀的限制，允许用户更灵活地定义表的主键。 [#40580](https://github.com/pingcap/tidb/issues/40580)
```

File: auto-increment.md [hunk 4]
```diff
@@ -351,7 +351,7 @@ CREATE TABLE t(a int AUTO_INCREMENT key) AUTO_ID_CACHE 1;
 
 目前在 TiDB 中使用 `AUTO_INCREMENT` 有以下限制：
 
-- 定义的列必须为主键或者索引的首列。
+- 对于 v6.6.0 及更早的 TiDB 版本，定义的列必须为主键或者索引的首列。
```
Comment (benmaoer) [inline auto-increment.md:354]:
```suggestion
- 对于 v6.6.0 及更早的 TiDB 版本，定义的列必须为主键或者索引前缀。
```

**Review by benmaoer — CHANGES_REQUESTED** [review]
微调为索引前缀

**Review by ran-huang — APPROVED** [review]

File: mysql-compatibility.md [hunk 5]
```diff
@@ -59,6 +59,8 @@ TiDB 高度兼容 MySQL 5.7 协议、MySQL 5.7 常用的功能及语法。MySQL
 
 自增 ID 详情可参阅 [AUTO_INCREMENT](/auto-increment.md)。
 
+- 对于 v6.6.0 及更早的 TiDB 版本，TiDB 的行为与 MySQL InnoDB  保持一致，要求自增列必须为主键或者索引前缀。 从 v7.0.0 开始，TiDB 移除自增列必须是索引或索引前缀的限制，允许用户更灵活地定义表的主键。 [#40580](https://github.com/pingcap/tidb/issues/40580)
```
Comment (Oreoxmt) [inline mysql-compatibility.md:62]:
```suggestion
- 对于 v6.6.0 及更早的 TiDB 版本，TiDB 的行为与 MySQL InnoDB 保持一致，要求自增列必须为主键或者索引前缀。从 v7.0.0 开始，TiDB 移除自增列必须是索引或索引前缀的限制，允许用户更灵活地定义表的主键。关于此更改的详细信息，请参阅 [#40580](https://github.com/pingcap/tidb/issues/40580)

自增 ID 详情可参阅 [AUTO_INCREMENT](/auto-increment.md)。
```

**Comment by ran-huang** [comment]
@benmaoer If you don't have other comments, please approve this PR.

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Pull request merged**